### PR TITLE
Header: disable level switching if lab loading

### PR DESCRIPTION
**Pull request by breville**
To make things simpler for a lab that supports switching levels without a page reload, this is a proposal to disable switching levels using the header bubbles if a level is currently loading. 

Without this change, we would have some issues with level switching initiated by pressing the header bubbles, in which we could end up with multiple requests (for level data, saved code, etc.) arriving asynchronously and potentially out of order.  The solution would be to wait for the right set of responses to arrive for the last-selected level, and might also involve waiting for all in-flight requests to be completed before settling on a level to show, and would be non-trivial to get right.

### CODE REVIEW

File: apps/src/code-studio/components/progress/LessonProgress.jsx [hunk 1]
```diff
@@ -71,6 +72,10 @@ class LessonProgress extends Component {
       }
     }
 
+    if (this.props.isLabLoading !== nextProps.isLabLoading) {
```
Comment (molly-moen) [inline apps/src/code-studio/components/progress/LessonProgress.jsx:75]: can you add a comment as to why this check is needed?

Comment (breville) [reply apps/src/code-studio/components/progress/LessonProgress.jsx:75]: Does the code make sense in the context of the `shouldComponentUpdate` function?  Basically everything here is just checking changed props that should result in doing a re-render...

Comment (molly-moen) [reply apps/src/code-studio/components/progress/LessonProgress.jsx:75]: I guess I'm wondering if there's ever a case where we would still want to render a change, even if the lab loading state hasn't changed. Maybe not? But worth explaining why.

Comment (breville) [reply apps/src/code-studio/components/progress/LessonProgress.jsx:75]: fwiw, we will still render a change for other props changing.  This just makes sure we also render a change if the loading state changes.

Comment (molly-moen) [reply apps/src/code-studio/components/progress/LessonProgress.jsx:75]: ah gotcha, makes sense!

**Review by molly-moen — APPROVED** [review]
🎉

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice! Agreed that this is a good mitigation for now, and indeed may be the behavior we want to have long term.

**Pull request merged**